mmdet3d/models/middle_encoders/sparse_encoder.py

```diff
@@ -46,19 +46,25 @@ class SparseEncoder(nn.Module):
             Default to False.
     """
 
-    def __init__(self,
-                 in_channels,
-                 sparse_shape,
-                 order=('conv', 'norm', 'act'),
-                 norm_cfg=dict(type='BN1d', eps=1e-3, momentum=0.01),
-                 base_channels=16,
-                 output_channels=128,
-                 encoder_channels=((16, ), (32, 32, 32), (64, 64, 64), (64, 64,
-                                                                        64)),
-                 encoder_paddings=((1, ), (1, 1, 1), (1, 1, 1), ((0, 1, 1), 1,
-                                                                 1)),
-                 block_type='conv_module',
-                 return_middle_feats=False):
+    def __init__(
+            self,
+            in_channels: int,
+            sparse_shape: List[int],
+            order: Optional[Tuple[str]] = ('conv', 'norm', 'act'),
+            norm_cfg: Optional[dict] = dict(
+                type='BN1d', eps=1e-3, momentum=0.01),
+            base_channels: Optional[int] = 16,
+            output_channels: Optional[int] = 128,
+            encoder_channels: Optional[Tuple[Tuple[int]]] = ((16, ), (32, 32,
+                                                                      32),
+                                                             (64, 64, 64),
+                                                             (64, 64, 64)),
+            encoder_paddings: Optional[Tuple[Tuple[int]]] = ((1, ), (1, 1, 1),
```

Comment:
This typehint is too long and ugly. Better like this alias: https://github.com/open-mmlab/mmdetection3d/blob/main/mmdet3d/models/backbones/pointnet2_sa_msg.py#L13
Please check all this similar type-hint about `channels` and `paddings`

Reply:
thanks for your  recommendations, I will modify these type-hint to the correct format.